### ISSUE TITLE
Can't create OAuth client ID

### ISSUE THREAD

**Issue by gtjamesa** [greghesp/assistant-relay#206]
**Describe the bug**

**This isn't a bug with `assistant-relay`**.

Just wondering if anybody else has encountered this and whether there is a solution? I have gone through the OAuth consent screen steps, right through to the "Return to dashboard" button. When trying to create an OAuth Client ID, I keep receiving a message saying that the "*Product* name" needs to be filled in, which it is ("App name").

![](https://i.imgur.com/hcYDQxV.png)
![](https://i.imgur.com/AqVs048.png)

**Comment by Conda86** [comment]
I get the exact same thing. Following this one.

**Comment by ryancasler** [comment]
I don't believe that the app name is allowed to have spaces in it.  Is it actually saving your changes to the consent screen?  Are you only putting in the app name and nothing else on the consent screen.  The same and email are literally the only things you need to fill out.

**Comment by gtjamesa** [comment]
I solved this by doing the following:

1. Go to [Firebase Console](https://console.firebase.google.com/) and select your project
2. Settings (top-left) -> Project Settings
3. Ensure your support email is set and set the "Public-facing name" field on this page. I used "Assistant Relay Client"
4. Save
5. Go back and configure your OAuth consent screen. You'll either have to now fill in the information or it will be [completed](https://i.imgur.com/MjDL8GB.png). As I had previously submitted the form and it wasn't working, it had approved the fields I had entered before
6. Proceed with the remaining steps for creating OAuth client ID
